chef/cookbooks/bind9/recipes/default.rb

```diff
@@ -182,19 +187,48 @@ def make_zone(zone)
   n = Node.load(n.name)
   cname = n["crowbar"]["display"]["alias"] rescue nil
   cname = nil unless cname && ! cname.empty?
+  base_name_no_net = n[:fqdn].chomp(".#{node[:dns][:domain]}")
+  alias_name_no_net = cname unless base_name_no_net == cname
+
   Chef::Recipe::Barclamp::Inventory.list_networks(n).each do |network|
     next unless network.address
-    base_name = n[:fqdn].chomp(".#{node[:dns][:domain]}")
-    alias_name = cname unless base_name == cname
-    unless network.name == "admin"
+    if network.name == "admin"
+      base_name = base_name_no_net
+      alias_name = alias_name_no_net
+    else
       net_name = network.name.gsub("_","-")
-      base_name = "#{net_name}.#{base_name}"
-      alias_name = "#{net_name}.#{alias_name}" if alias_name
+      base_name = "#{net_name}.#{base_name_no_net}"
+      alias_name = "#{net_name}.#{alias_name_no_net}" if alias_name_no_net
     end
     cluster_zone[:hosts][base_name] = Mash.new
     cluster_zone[:hosts][base_name][:ip4addr] = network.address
     cluster_zone[:hosts][base_name][:alias] = alias_name if alias_name
   end
+
+  # Also set DNS name with temporary DHCP address for discovered nodes
+  if n[:state] == "discovered" &&
+    !cluster_zone[:hosts].key?(base_name_no_net) &&
+    Chef::Recipe::Barclamp::Inventory.get_network_by_type(n, "admin").nil?
+    address = n[:ipaddress]
+    time = n[:ohai_time]
+    use_temporary = true
+
+    unless temporary_dhcp[address].nil?
+      temp_time, temp_base_name, temp_alias_name = temporary_dhcp[address]
+      use_temporary = false if time < temp_time
+    end
+
+    if use_temporary
+      temporary_dhcp[address] = [time, base_name_no_net, alias_name_no_net]
+    end
+  end
+end
+
+temporary_dhcp.each_pair do |address, value|
+  time, base_name, alias_name = value
```

Comment:
Lint/UselessAssignment: Useless assignment to variable - `time`. Use `_` or `_time` as a variable name to indicate that it won't be used. (https://github.com/bbatsov/ruby-style-guide#underscore-unused-vars)